core/fetcher.go

```diff
@@ -0,0 +1,87 @@
+package core
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/celestiaorg/celestia-core/types"
+	"github.com/celestiaorg/celestia-node/service/block"
+)
+
+const newBlockSubscriber = "NewBlock/Events"
+
+var newBlockEventQuery = types.QueryForEvent(types.EventNewBlock).String()
+
+type BlockFetcher struct {
+	client Client
+
+	mux        sync.Mutex
+	newBlockCh chan *block.Raw
```

Comment:
Can you add a comment on why the mutex is necessary on top of the `mux` field?

Reply:
Also (if it is actually necessary), I'd recommend naming the mutex with to what it locks (e.g. newBlockChLock).

Reply:
why `new` though? Does not make sense to me.

Reply:
https://github.com/celestiaorg/celestia-node/pull/59#issuecomment-920856968

Reply:
Ok, maybe newBlocks then?

Reply:
I usually like to use the `___Ch` naming convention for channels. Is that okay with you? Or should we use some other naming scheme and keep it consistent.

Reply:
I am ok with `_Ch`. I meant not `newBlock`, but `newBlocks`, but I don't think strongly about this